### PR TITLE
Refining Build Scripts/Docs for Clean Ubuntu 18.04 Installation

### LINKED ISSUE #154: Building the repo on Ubuntu 18.04 desktop
@isherman 

Following the development guide is going well. I few minor issues have cropped up, mostly due to user error:

- Did not have virtualenv installed the first time I ran `make`
- The second time I tried the build, the error message said that the bootstrap recipe failed, due to some files already existing in  `/tmp`
  - resolved by running `sudo rm -r /tmp/node_exporter-1.0.1.linux-amd64* /tmp/prometheus-2.21.0.linux-amd64*`
- The last time I tried the build, this error resulted:
```code
     ERROR: Command errored out with exit status 1:
     command: /home/collin/farm-ng/tractor/env/bin/python2 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-p2vNjK/websockets/setup.py'"'"'; __file__='"'"'/tmp/pip-install-p2vNjK/websockets/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-mv_RWW
         cwd: /tmp/pip-install-p2vNjK/websockets/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-p2vNjK/websockets/setup.py", line 1, in <module>
        import pathlib
    ImportError: No module named pathlib
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

- After installing `python-pathlib` the same error resulted.
  - The process that produces this error is "Collecting websockets"
  - The first time (before installing python-pathlib) it said "Downloading websockets-5.0.1.tar.gz" but now says "Using cached websockets-5.0.1.tar.gz"

@mabitter

### PR DESCRIPTION
**Pull request by collinbrake**
closes #154

### CODE REVIEW

**Comment by collinbrake** [comment]
@isherman Should I update the dev guide by replacing `git clone git@github.com:USERNAME/tractor ~/tractor` with `git clone -r git@github.com:USERNAME/tractor ~/tractor`, using recursive clone to snag the submodules?

**Comment by isherman** [comment]
> @isherman Should I update the dev guide by replacing `git clone git@github.com:USERNAME/tractor ~/tractor` with `git clone -r git@github.com:USERNAME/tractor ~/tractor`, using recursive clone to snag the submodules?

Yes, for sure, thank you. 

It's a nit, but let's use `--recursive` instead of `-r` -- using the full flag name in documentation / examples / bash scripts helps new developers understand what's going on.

**Review by isherman — APPROVED** [review]
Approving, but let's wait for CI to go green before merging (just triggered a re-start of CI, since it looked like the failure might have been an ephemeral issue an HTTP endpoint).

**Comment by collinbrake** [comment]
@isherman We may want to wait on this - the build was successful, but it doesn't look like the executables from the cpp code exist in the build directory yet, and the test for python executables doesn't work yet.

File: dev.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM ubuntu:18.04
```
Comment (isherman) [inline dev.Dockerfile:1]: Can you move this file into `docker/`

File: bootstrap-venv.sh [hunk 2]
```diff
@@ -11,15 +11,19 @@ export FARM_NG_ROOT=$( cd "$( dirname "${SOURCE}" )" >/dev/null 2>&1 && pwd )
...
+pip3 install -r $FARM_NG_ROOT/requirements.txt
 
 # Go
 FARM_NG_GOPATH=$FARM_NG_ROOT/env/go
 mkdir -p $FARM_NG_GOPATH
 export GOPATH=$FARM_NG_GOPATH:$GOPATH
 go get -u github.com/golang/protobuf/protoc-gen-go
 go get -u github.com/twitchtv/twirp/protoc-gen-twirp
+
+pip3 install wheel
```
Comment (isherman) [inline bootstrap-venv.sh:28]: Looks like this duplicates the above lines

**Review by isherman — CHANGES_REQUESTED** [review]
added a couple cleanup requests

**Comment by collinbrake** [comment]
@isherman 

1. Should I squash any of these commits before the merge? It looks like some of your commits are mixed in - not sure how that happened, since I pulled them from this repository.
1. To update the dev wiki, do I clone that separately and edit or should it be included in this PR?

**Comment by collinbrake** [comment]
@isherman OK, will look into the cleanups.

**Review by isherman — APPROVED** [review]

**Pull request merged**